source/slang/slang-shader-cache-index.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "../../slang.h"
+
+#include "../core/slang-string.h"
+#include "../core/slang-linked-list.h"
+#include "../core/slang-dictionary.h"
+
+namespace Slang
+{
+    class ShaderCacheIndex
```

Comment:
There should be a `ShaderCacheIndexEntry` class with `Digest dependencyBasedDigest;` and `Digest astBasedDigest` fields.